db/seeds.rb

```diff
@@ -1,7 +1,135 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
-#   Character.create(name: "Luke", movie: movies.first)
+# Clear the database to prevent duplicate records
+Contact.destroy_all
+User.destroy_all
+Friendship.destroy_all
+Interaction.destroy_all
+Medium.destroy_all
+Note.destroy_all
+Reminder.destroy_all
+UserSetting.destroy_all
+
+# Users
+5.times do |i|
+  User.create!(
+    email: Faker::Internet.email,
+    encrypted_password: Devise::Encryptor.digest(User, 'password'),
+    username: "user#{i}",
+    password: "123456",
+    first_name: Faker::Name.first_name,
+    middle_name: Faker::Name.middle_name,
+    last_name: Faker::Name.last_name,
+    nickname: Faker::Name.first_name,
+    emoji: ["☀️", "🌈", "🍿", "🐝", "🎈"].sample
+  )
+end
+
+User.create(
+  username: "emmvs", 
+  email: "emma@test.com", 
+  password: "123456", 
+  first_name: "Emma", 
+  middle_name: "Anna Agneta", 
+  last_name: "Rünzel",
+  nickname: "Emmsi",
+  emoji: "☀️"
+)
+
+User.create(
+  username: "xxjmsxx", 
+  email: "josh@test.com", 
+  password: "123456", 
+  first_name: "Joshua", 
+  middle_name: nil, 
+  last_name: "Smith",
+  nickname: "Josh",
+  emoji: "🍰"
+)
+
+User.create(
+  username: "santiagosan93", 
+  email: "santi@test.com", 
+  password: "123456", 
+  first_name: "Santiago", 
+  middle_name: nil, 
+  last_name: "Sanchez",
+  nickname: "Santi",
+  emoji: "🐘"
+)
+
+users = User.all
+
+# Contacts
+users.each do |user|
+  5.times do
+    Contact.create!(
+      name: Faker::Name.name,
+      relationship: ['friend', 'family', 'colleague'].sample,
+      email: Faker::Internet.email,
+      phone_number: Faker::PhoneNumber.cell_phone,
+      address: Faker::Address.full_address,
+      social_media_handle: "@#{Faker::Internet.username}",
+      user: user
+    )
+  end
+end
+
+contacts = Contact.all
+
+# Friendships
+users.each_with_index do |user, index|
+  friend = users.where.not(id: user.id).sample
+  Friendship.create!(
+    status: [0, 1, 2].sample, # assuming 0: requested, 1: accepted, 2: declined
+    requested_at: Time.now,
+    responded_at: [nil, Time.now].sample,
+    user: user,
+    friend_id: friend.id
+  )
+end
+
+# Media
+users.each do |user|
+  Medium.create!(
+    name: ['Email', 'Phone Call', 'Social Media', 'In Person'].sample,
+    user: user
+  )
+end
+
+# Interactions
+contacts.each do |contact|
+  Interaction.create!(
+    interaction_date: Faker::Date.between(from: 2.days.ago, to: Date.today),
+    user: contact.user,
+    contact: contact,
+    medium_id: [1, 2, 3].sample # Assuming these IDs exist in your media table
```

Comment:
```suggestion
    medium_id: Medium.all.pluck(:id).sample # Assuming these IDs exist in your media table
```
[`#pluck`](https://apidock.com/rails/ActiveRecord/Calculations/pluck)

Reply:
COOOL!!!!!